jerry-core/api/jerry.c

```diff
@@ -847,8 +840,6 @@ jerry_value_get_type (const jerry_value_t value) /**< input value to check */
       break;
     }
   }
-
-  return JERRY_TYPE_NONE;
```

Comment:
It is strange the checkers does not complain (for the the UNREACHABLE is just a function call). We should move this up to the default case.